### ISSUE TITLE
Windows 64 Bit release as tar.xz download

### ISSUE THREAD

**Issue by bebuch** [llvm/llvm-project#62492]
- https://github.com/llvm/llvm-project/releases/tag/llvmorg-16.0.2

Currently you provide Windows Binaries as executable Installer.

Due to internal company restrictions I cannot run an installer. Can you please provide a tar.xz version of the Windows 64 bit binaries?

I want to automate the "installation" together with some other programs via CMake. LLVM is currently the only one without a ZIP file download.

I suppose that a workaround would be possible using 7-Zip. But for that you would have to install 7-Zip and they also only provide an installer. (And a version as .7z archive, which is pointless).

**Comment by bebuch** [comment]
tl;dr: Looks like a workaround with 7-Zip is not possible.

I found out that CMake supports unpacking 7z-Archives since 3.1. So I was able to provide 7-Zip by downloading [there 7z-Archrive](https://7-zip.org/download.html) (7-Zip Extra: standalone console version, 7z DLL, Plugin for Far Manager). Also I found the Windows x86-32 exe file there which works without extracting anything.

Unfortunately, the console versions of 7-Zip have a reduced set of supported formats. The exe installer format used for LLVM is not included. This does work with the graphical version of 7-Zip only. So a manual step would be required and it is impossible to use it in CI pipelines.

**Comment by bebuch** [comment]
After two weeks of research, I am really frustrated. There is simply no command line way to extract the data from NSIS installers. I can't even run the file as that requires admin privileges. 7-Zip is the only thing that worked, but only in the GUI version. That can't be used for automation, so you always need a manual step.

I beg you, please provide a normal ZIP file or similar for Windows binaries.

**Comment by zmodem** [comment]
> Unfortunately, the console versions of 7-Zip have a reduced set of supported formats. The exe installer format used for LLVM is not included. This does work with the graphical version of 7-Zip only. So a manual step would be required and it is impossible to use it in CI pipelines.

What version of 7-Zip are you using? The command-line version works for me:

```
>7z x \src\llvm_package_17.0.0-rc4\build64\LLVM-17.0.0-rc4-win64.exe

7-Zip 19.00 (x64) : Copyright (c) 1999-2018 Igor Pavlov : 2019-02-21

Scanning the drive for archives:
1 file, 331079640 bytes (316 MiB)

Extracting archive: \src\llvm_package_17.0.0-rc4\build64\LLVM-17.0.0-rc4-win64.exe
--
Path = \src\llvm_package_17.0.0-rc4\build64\LLVM-17.0.0-rc4-win64.exe
Type = Nsis
Physical Size = 331079640
Method = LZMA:25
Solid = +
Headers Size = 392228
Embedded Stub Size = 116736
SubType = NSIS-3 Unicode BadCmd=37

Everything is Ok

Files: 468
Size:       2821943858
Compressed: 331079640
```

**Comment by bebuch** [comment]
What you are using is 7zip from the installable 7zip package.

The command line only versi, which can be used without install is:

- https://7-zip.org/a/7z2301-extra.7z

or

- https://7-zip.org/a/7zr.exe

But it doesn't support Nsis:

```
> 7z2301-extra\x64\7za.exe x LLVM-16.0.6-win64.exe

7-Zip (a) 23.01 (x64) : Copyright (c) 1999-2023 Igor Pavlov : 2023-06-20

Scanning the drive for archives:
1 file, 307648319 bytes (294 MiB)

Extracting archive: LLVM-16.0.6-win64.exe
ERROR: LLVM-16.0.6-win64.exe
Cannot open the file as archive


Can't open as archive: 1
Files: 0
Size:       0
Compressed: 0
```

What you are using is:

- https://7-zip.org/a/7z2301-x64.msi

or

- https://7-zip.org/a/7z2301-x64.exe

Both require admin rights to be installed. So I have the same problem as with the LLVM installer itself.
